azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/utils/Utils.java

```diff
@@ -369,4 +371,53 @@ public FileVisitResult visitFileFailed(Path file, IOException exc) {
         }
         return tarFilePath;
     }
+
+    public static boolean isSpringBootProject(Path projectPath) {
+        try {
+            // Check if `pom.xml` exists and contains Spring Boot references
+            Path pomPath = projectPath.resolve("pom.xml");
+            if (Files.exists(pomPath) &&
+                (containsText(pomPath, "org.springframework.boot") ||
+                    containsText(pomPath, "spring-boot-maven-plugin"))) {
```

Comment:
Can we use more accurate detection? For example check for XML elements rather than using raw text match?